FlowCrypt/Functionality/Services/GoogleUserService.swift

```diff
@@ -193,13 +193,17 @@ extension GoogleUserService: UserServiceType {
 extension GoogleUserService {
 
     private func makeAuthorizationRequest(scopes: [GoogleScope]) -> OIDAuthorizationRequest {
-        OIDAuthorizationRequest(
+        var additionalParameters = ["include_granted_scopes": "true"]
+        if let currentUserEmail = currentUserEmail {
+            additionalParameters["login_hint"] = currentUserEmail
+        }
```

Comment:
Please pass this as a parameter. to the method I'm trying to avoid this pattern of "methods getting their own stuff internally", instead values should be always dependency-injected through method parameters going forward.

Reply:
Then what's the usecase of `currentUserEmail` variable in GoogleUserService?

Reply:
I really, really want to get rid of it, completely if I can. Or mostly, if I cannot remove it completely.

But because it's still used across 30-40% of the app, it's a slow process. But it will have to involve not adding more usages :-)

Sorry this wasn't documented. Please add a comment on that property that we are trying to deprecate it. Actually, if there is a swift feature that you could mark it with, to produce warnings on all usages, that would be even better.

Reply:
The right way to get current user, under the newer design intention, is to get it from the VC which has it as an input from the global router.

From VC, it goes through method parameters all the way down to wherever it's needed.

Reply:
Ok, got it

Reply:
![image](https://user-images.githubusercontent.com/77344191/159927770-ec336b91-723b-4ee7-9091-37ddd2df8cd4.png)


What do you think?

Reply:
@tomholub

Reply:
looks good